src/Microsoft.TemplateEngine.Cli/HelpAndUsage/HelpForTemplateResolution.cs

```diff
@@ -319,32 +319,33 @@ private static void DisplayPartialNameMatchLanguageAndContextProblems(string tem
             if (templateResolutionResult.IsNoTemplatesMatchedState || templateResolutionResult.UsingPartialMatches)
             {
                 ShowNoTemplatesFoundMessage(templateName, templateLanguage, context);
+                Reporter.Error.WriteLine();
                 return;
             }
 
             bool anythingReported = false;
-
-            foreach (IReadOnlyList<ITemplateMatchInfo> templateGroup in templateResolutionResult.ContextProblemMatchGroups)
-            {
-                // all templates in a group should have the same context & name
-                if (templateGroup[0].Info.Tags != null && templateGroup[0].Info.Tags.TryGetValue("type", out ICacheTag typeTag))
+            int partialTemplatesMatchCount = templateResolutionResult.ContextProblemMatchGroups.Count(templateGroup =>
                 {
-                    MatchInfo? matchInfo = WellKnownSearchFilters.ContextFilter(context)(templateGroup[0].Info);
-                    if ((matchInfo?.Kind ?? MatchKind.Mismatch) == MatchKind.Mismatch)
+                    // all templates in a group should have the same context & name
+                    if (templateGroup[0].Info.Tags != null && templateGroup[0].Info.Tags.TryGetValue("type", out ICacheTag typeTag))
```

Comment:
nit: Since the `typeTag` argument is not used, this could be calling `ContainsKey()` instead of `TryGetValue()`.